flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -155,6 +163,15 @@ public void run() {
 					// we can close this consumer thread once we've reached the end of the subscribed shard
 					break;
 				} else {
+					if (fetchIntervalMillis != 0) {
```

Comment:
Yup. Adding the validation to `KinesisConfigUtil.validateConfiguration()`, like the other values.